mypy/options.py

```diff
@@ -93,7 +93,7 @@ def __init__(self) -> None:
         self.junit_xml = None  # type: Optional[str]
 
         # Per-module options (raw)
-        self.per_module_options = {}  # type: Dict[str, Dict[str, object]]
+        self.per_module_options = {}  # type: Dict[Pattern[str], Dict[str, object]]
```

Comment:
BTW it's odd to me that this is a Dict.  It means that later identical entries clobber earlier entries entirely.  And since dictionary traversal order is not fixes it means you can rely on any kind of ordering between entries.  For example:

`
[mypy-foo.*]
ignore_errors = True

[mypy-foo.bar]
ignore_errors = False
```

Whether foo.bar is ignored or not depends on how dictionary traversal is done.  (Hmm. Maybe I should write separate bug.)